test/functional/p2p_segwit.py

```diff
@@ -1418,16 +1418,24 @@ def test_segwit_versions(self):
         temp_utxo.pop()  # last entry in temp_utxo was the output we just spent
         temp_utxo.append(UTXO(tx2.sha256, 0, tx2.vout[0].nValue))
 
-        # Spend everything in temp_utxo back to an OP_TRUE output.
+        # Spend everything in temp_utxo into an segwit v1 output.
         tx3 = CTransaction()
         total_value = 0
         for i in temp_utxo:
             tx3.vin.append(CTxIn(COutPoint(i.sha256, i.n), b""))
             tx3.wit.vtxinwit.append(CTxInWitness())
             total_value += i.nValue
         tx3.wit.vtxinwit[-1].scriptWitness.stack = [witness_program]
-        tx3.vout.append(CTxOut(total_value - 1000, CScript([OP_TRUE])))
+        tx3.vout.append(CTxOut(total_value - 1000, script_pubkey))
         tx3.rehash()
+
+        # First we test this transaction against fRequireStandard=true node
+        # making sure the txid is added to the reject filter
+        self.std_node.announce_tx_and_wait_for_getdata(tx3)
+        test_transaction_acceptance(self.nodes[1], self.std_node, tx3, with_witness=True, accepted=False, reason="bad-txns-nonstandard-inputs")
```

Comment:
If I remove this `reason` parameter, then this test passes on master (i.e. I don't think the line below actually tests any functional change).

Reply:
Good catch. Increasing the wait to 5 seconds seems to fix it, probably because of INV delay logic? 2 seconds isn't enough time for the INV to get sent out by the node.

Reply:
I wonder if having a negative log line is better for each negative test case so we don't have to guess what the time required is.

Reply:
You could use `setmocktime` to move time forward and force the node to send out getdata requests.

Reply:
Ok, I'd suggest just doing 5 seconds for now for both instances of `announce_tx_and_wait_for_getdata` `success=False` in this test.

Reply:
Changed this to 5 seconds.